syncer/syncer.go

```diff
@@ -753,11 +753,16 @@ func (s *Syncer) addJob(job *job) error {
 		// save job's current pos for DML events
 		for i := range job.sourceSchemas {
 			if len(job.sourceSchemas[i]) > 0 {
-				s.saveTablePoint(job.sourceSchemas[i], job.sourceTables[i], job.location)
+				s.saveTablePoint(job.sourceSchemas[i], job.sourceTables[i], job.currentLocation)
 			}
 		}
 	}
 
+	failpoint.Inject("FlushCheckpointExit", func() {
```

Comment:
in many places, we call `s.flushCheckPoints` directly, but they will not trigger this `FlushCheckpointExit`, how about giving it another name or move it into `s.flushCheckPoints`?